rust/db_handling/src/identities.rs

```diff
@@ -290,32 +321,65 @@ fn import_all_addrs_v1<P: AsRef<Path>>(
-                        derivation_path,
-                        &NetworkSpecsKey::from_parts(
-                            &derived_key.genesis_hash,
-                            &derived_key.encryption,
-                        ),
-                        &db_path,
-                    );
-                }
+    for addr in seed_derived_keys.iter_mut() {
+        for mut derived_key in addr.derived_keys.iter_mut() {
+            let path = derived_key.derivation_path.clone().unwrap_or_default();
+            let seed_name = match addr.multisigner {
+                MultiSigner::Sr25519(p) => sr25519_signers.get(&p),
+                MultiSigner::Ed25519(p) => ed25519_signers.get(&p),
+                MultiSigner::Ecdsa(p) => ecdsa_signers.get(&p),
+            };
+            if seed_name.is_none() {
+                // unknown seed, skipping
+                continue;
             }
+            let seed_name = seed_name.unwrap();
+            let seed_phrase = seeds.get(seed_name.as_str()).unwrap();
+            // create fixed-length string to avoid reallocation
+            let mut full_address = String::with_capacity(seed_phrase.len() + path.len());
+            full_address.push_str(seed_phrase);
+            full_address.push_str(&path);
+
+            let multisigner_pwdless = match derived_key.encryption {
+                Encryption::Ed25519 => match ed25519::Pair::from_string(&full_address, None) {
+                    Ok(a) => {
+                        full_address.zeroize();
+                        MultiSigner::Ed25519(a.public())
+                    }
+                    Err(e) => {
+                        full_address.zeroize();
+                        return Err(Error::SecretStringError(e));
+                    }
+                },
+                Encryption::Sr25519 => match sr25519::Pair::from_string(&full_address, None) {
+                    Ok(a) => {
+                        full_address.zeroize();
+                        MultiSigner::Sr25519(a.public())
+                    }
+                    Err(e) => {
+                        full_address.zeroize();
+                        return Err(Error::SecretStringError(e));
+                    }
+                },
+                Encryption::Ecdsa | Encryption::Ethereum => {
+                    match ecdsa::Pair::from_string(&full_address, None) {
+                        Ok(a) => {
+                            full_address.zeroize();
+                            MultiSigner::Ecdsa(a.public())
+                        }
+                        Err(e) => {
+                            full_address.zeroize();
+                            return Err(Error::SecretStringError(e));
+                        }
+                    }
+                }
+            };
+            let multisigner =
+                base58_or_eth_to_multisigner(&derived_key.address, &derived_key.encryption)?;
+            derived_key.has_pwd = Some(multisigner_pwdless != multisigner);
```

Comment:
so essentially here you have to go from multisigner_pwdless to address and compare it to `derived_key.address`, not the other way around. This way you in case of Ethereum will go from ecdsa multisigner -> to 20-byte hex address.

Reply:
Fixed